sympy/functions/elementary/exponential.py

```diff
@@ -1009,15 +1009,22 @@ def coeff_exp(term, x):
             while s.is_Order:
                 n += 1
                 s = arg.nseries(x, n=n, logx=logx)
-        a, b = s.removeO().leadterm(x)
+        if logx.has(x) and logx != log(x):
+            a, b = arg.subs(log(x), logx).as_leading_term(x), S.Zero
+        else:
+            a, b = s.removeO().leadterm(x)
         p = cancel(s/(a*x**b) - 1).expand().powsimp()
         if p.has(exp):
             p = logcombine(p)
         if isinstance(p, Order):
             n = p.getn()
         _, d = coeff_exp(p, x)
         if not d.is_positive:
-            return log(a) + b*logx + Order(x**n, x)
+            res = log(a) + b*logx
+            if (res.subs(logx, log(x)) == self
+                or res.subs(-logx, -log(x)) == self):
```

Comment:
Actually another test from `test1_latest` which fails , is due to this sort of handling but in `mul.py` actually lands up on this 
```
>>> expr = x**2*_eis(-2*log(1/x))/2
>>> expr._eval_nseries(x, n = 6, logx = -p)
x**2*_eis(-2*p)/2 + O(x**6)
```
But the order term shouldn't be there 
Now this is because I've added a simple case like 
```
        if res != self:
            if res.subs(logx, log(x)) == self:
                return res
```
But here `res  = x**2*_eis(-2*p)/2` and `logx = -p` so we get the substituted expression as `x**2*_eis(2*log(x))/2` which is correct but not in the form of self (uses log(1/x) so yeah here the case gets more weird , we would need to have something like 
```
         if res != self:
+            if (res.subs(logx, log(x)) == self
+                or res.subs(-logx, -log(x)) == self
+                or res.subs(-logx, log(1/x)) == self):
                         return res;
```
Then we get the correct result

Reply:
We cannot possibly have substitutions for many different expressions like `log(1/x)`. Instead, we should probably first try to force `self` into a more canonical form by something like `expand_log` and then substitute for `logx` (or `-logx` if `logx` is negative).
```
In [11]: res = x**2*_eis(-2*p)/2  

In [12]: logx = -p

In [13]: res.subs(-logx, -log(x))
Out[13]: 
 2               
x ⋅_eis(2⋅log(x))
─────────────────
        2        

In [14]: expr = x**2*_eis(-2*log(1/x))/2

In [17]: expand_log(expr, force=True)
Out[17]: 
 2               
x ⋅_eis(2⋅log(x))
─────────────────
        2
```

Reply:
Thanks for this, wasn't aware of `expand_log`, I was just about to commit using the following , then saw your message
``
            expr = expr.expand(**dict(deep=True, log=True, mul=False, power_exp=False,
                  power_base=False, multinomial=False, basic=False, force=False,
                  factor=False)
``

Reply:
I think, we can try with the above snippet anutosh posted, with `force=True`, and see if that works. `expand_log` can be more expensive.

Reply:
The only case which we would be missing out here would be the ones having `log(1/x)`.For eg As of now we have the following 
```
>>> log(log(1/x))._eval_nseries(x, n= 3, logx=-p)  
log(p) + O(x**3)    # order term is unwanted
 ```
And let's say we stick to something like 
```
        if not d.is_positive:
            res = log(a) + b*logx
            expr = self.expand(**dict(deep=True, log=True, mul=False,
                    power_exp=False, power_base=False, multinomial=False,
                    basic=False, force=True, factor=False))
            if (res.subs(logx, log(x)) == expr
                or res.subs(-logx, -log(x)) == expr):
                return res
```
where `res = log(p)`, `logx= -p` . We would miss out on cases having log(1/x) as shown below 
```
>>> expand_log(log(log(1/x)), force=True)
log(log(x)) + I*pi
```
This sort of implementation would have liked the answer as `log(-log(x))` rather than `log(log(x)) + I*pi`

Reply:
I've addressed this now , everything works fine now . Hence resolving (failing tests not due to the latest commit , due to some other warnings I see)